src/readers/morphologyHDF5.cpp

```diff
@@ -78,9 +78,165 @@ namespace morphio {
 namespace readers {
 namespace h5 {
 
+
+namespace detail {
+
+// Note, that `hsize_t` might be `uint64_t` which could with `uint64_t` or
+// `unsigned long long` depending on the HDF5 version and platform. Foremost,
+// this is an issue with code not compiling, e.g. if one defines one overload
+// for both `size_t` and `hsize_t` then this will break on platforms where they
+// are the same; if one doesn't it'll break on platforms were it isn't. To
+// avoid going into template metaprogramming, this function allows copying to
+// the right type. Usually `sizes` is small, i.e. 1-2 elements small.
+static std::vector<size_t> convert_to_size_t(const std::vector<hsize_t>& sizes) {
+    return std::vector<size_t>(sizes.begin(), sizes.end());
+}
+
+static void check_dimensions_match(const std::vector<size_t>& actualDimensions,
+                                   const std::vector<size_t>& expectedDimensions,
+                                   const std::string& datasetName) {
+    if (actualDimensions.size() != expectedDimensions.size()) {
+        throw RawDataError("dataset: " + datasetName + ": bad number of dimensions.");
+    }
+
+    for (size_t k = 0; k < actualDimensions.size(); ++k) {
+        size_t expected = expectedDimensions[k];
+        size_t actual = size_t(actualDimensions[k]);
```

Comment:
is the `size_t` needed?
```suggestion
        size_t actual = actualDimensions[k];
```

Reply:
not anymore.